src/main/java/org/opensearch/sdk/ExtensionsRunner.java

```diff
@@ -181,6 +199,7 @@ InitializeExtensionsResponse handleExtensionInitRequest(InitializeExtensionsRequ
         } finally {
             // After sending successful response to initialization, send the REST API and Settings
             setOpensearchNode(opensearchNode);
+            setExtensionNode(extensionInitRequest.getExtension());
```

Comment:
What's the reason to set the node here since the node is already present in the `extensionInitRequest`?

Reply:
We set the node here and pass the DiscoveryExtension object back to the SDK. This is then later passed to OpenSearch within the `AddSettingUpdateConsumerRequest` to enable the callback consumer to send a transport request to the extension node that registered the setting consumer

Reply:
Can't we directly pass `extensionInitRequest.getExtension()` to `AddSettingUpdateConsumerRequest`? In the future, we need to pass an entire list of extenionsNode so setting a specific node doesn't look good here.

Reply:
The `InitializeExtensionsRequest` object is only available within the scope of `handleExtensionInitRequest` method, so in order to get the node when creating an `AddSettingUpdateConsumerRequest`, it needs to be available within the SDK. `AddSettingUpdateConsumerRequests` will not be initialized within `handleExtensionInitRequest`, so we cannot directly pass the extensionNode from the `extensionInitRequest` directly. Instead, this will be sent during the create-components workflow, after extensions have registered their setting consumers. All `setExtensionNode(extensionInitRequest.getExtension())` does is set the `extensionNode` in order to configure callback requests to this extension